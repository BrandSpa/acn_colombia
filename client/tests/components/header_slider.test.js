import React from "react";
import { shallow, mount } from "enzyme";
import sinon from 'sinon';
import HeaderSlider from "../../components/header_slider";

describe("header slider component", () => {
	it('should render as expected', () => {
		let wrapper = shallow(<HeaderSlider />);
		expect(wrapper.getNodes()).toMatchSnapshot();
	})

	it('should change slide to next', () => {
		let wrapper = shallow(<HeaderSlider slides={[{title: 'acn pope'}, {title: 'acn pope'}, {title: 'acn pope'}]} />);
		wrapper.instance().nextSlide();
		wrapper.instance().nextSlide();
		expect(wrapper.state()).toEqual({currentSlide: 2, left: '-200%'});
	})

	it('should change slide to prev', () => {
		let wrapper = shallow(<HeaderSlider slides={[{title: 'acn pope'}, {title: 'acn pope'}, {title: 'acn pope'}]} />);
		wrapper.instance().nextSlide();
		wrapper.instance().nextSlide();
		wrapper.instance().prevSlide();
		expect(wrapper.state()).toEqual({currentSlide: 1, left: '-100%'});
	})

	it('should show video modal', () => {
			let wrapper = mount(<HeaderSlider slides={[{title: 'acn pope', 'isvideo': true}]} />);
			wrapper.find('.slider__slide').simulate('click');
			expect( wrapper.find('.modal--show').length).toEqual(1);
	})

});
