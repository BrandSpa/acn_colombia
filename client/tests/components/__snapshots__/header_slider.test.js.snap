// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header slider component should render as expected 1`] = `
Array [
  <div
    className="slider"
    style={
        Object {
            "height": "auto",
          }
    }
>
    <div
        className="slider__viewport"
        style={
            Object {
                "height": "auto",
                "left": "0",
                "width": "0%",
              }
        }
    >
        
    </div>
    
</div>,
]
`;
